src/FormFileLabel.js

```diff
@@ -40,7 +40,7 @@ const FormFileLabel = React.forwardRef(
         ref={ref}
         htmlFor={htmlFor || controlId}
         className={classNames(className, bsPrefix)}
-        data-browse={buttontext}
+        data-browse={buttonText}
```

Comment:
what uses this data attribute?

Reply:
The data attribute is used to set the text of the "button" (really an ::after pseudo element) in the custom file uploader label.

It can then be used to translate the text in the "button" (either hardcoded or with i18next or similar). It can also be done via CSS but that falls outside the scope of what we are doing here :)

Bootstrap docs here: https://getbootstrap.com/docs/4.4/components/forms/#translating-or-customizing-the-strings-with-html

<img width="1283" alt="Screenshot 2020-03-11 at 00 12 34" src="https://user-images.githubusercontent.com/322759/76367356-1aa63600-632d-11ea-92e7-93e4c5ad8259.png">

Reply:
interesting... i guess this is the JS that TWBS pulls in?

Reply:
Actually it is not. The JS they add is only used for displaying the uploaded files name within the custom file uploader. See the end for what the JS does :)

**What the CSS does/how it works**
The data-attribute to button text is a bunch of CSS selector magic and pseudo-elements
Looking att TWBS and how they do it (and how we replicate it with the buttonText prop) is that on the `<label>` element you set the data-browse attribute to a string like so:
`<label class="custom-file-label" for="customFileLangHTML" data-browse="My Custom Button Text">My Regular Label Text</label>`

And with that set the CSS magic works:
```
.custom-file-input~.custom-file-label[data-browse]::after {
    content: attr(data-browse);
}
```

This is why the order of the label and input elements becomes important. When we have an element with the class `.custom-file-input` followed by an element with the class `.custom-file-label` with the data attribute `data-browse` set an `::after` pseudo-element is created. And that pseudo-elements content is set to the data-attribute by using `content: attr(data-browse)`

All according to how `attr()` is supposed to be used by the spec. More over at MDN: https://developer.mozilla.org/en-US/docs/Web/CSS/attr

Totally done without any JS involved at all. The same goes for the `lang` prop that I added in another commit, useable without any JS involved at all.


_**This is what the JS does**_
**With JS**
Choose file becomes [filename]
![customFileJS](https://user-images.githubusercontent.com/322759/76392681-51526f80-6372-11ea-8b3a-367a3f432088.gif)

**Without JS**
Choose file does not change (but the value for the underlying input element does)
![customFileNoJS](https://user-images.githubusercontent.com/322759/76392890-c0c85f00-6372-11ea-9338-0c211f021d21.gif)

Reply:
@jquense are you okay with the new data-browse attribute and the overview of how it works above? Can we resolve this? :)